src/org/thoughtcrime/securesms/ApplicationPreferencesActivity.java

```diff
@@ -122,7 +122,7 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
       DynamicTheme.setDefaultDayNightMode(this);
       recreate();
     } else if (key.equals(Prefs.LANGUAGE_PREF)) {
-      recreate();
+      dynamicLanguage.onResume(this);
```

Comment:
nice thing is that this results in easy maintenance - exactly the same is done for other activities on the language change